l10n_it_fiscalcode/migrations/10.0.1.0.1/post-migrate.py

```diff
@@ -15,4 +15,5 @@ def migrate(cr, version):
         partners = env['res.partner'].search(
             [('is_company', '=', True)])
         for partner in partners:
-            partner._commercial_sync_to_children()
+            partner = env['res.partner'].browse(partner.id)
```

Comment:
partner è già un browse record

Reply:
Queste due righe di codice mi hanno incuriosito :laughing: 

In effetti scrivendo:
```
       partners = env['res.partner'].search(
            [('is_company', '=', True)])
```
i recordset `partners` sono pronti per essere prefetchati: appena un campo di uno dei record viene letto, tutti gli altri campi di tuti gli altri record vengono prefetchati e quindi letti.
Il prefetch si attiva quando invoco `_commercial_sync_to_children()` perche' in questo metodo vengono letti alcuni campi del partner. Inoltre `def _commercial_sync_to_children()` richiama se stessa ricorsivamente per ciascun child del partner. Quindi se `partners` ha parecchi record e invocando `_commercial_sync_to_children()` su ognuno dei partner, il meccanismo di prefetch comporta un rallentamento generale.

Invece scrivendo:
```
partner = env['res.partner'].browse(partner.id)
```
e poi invocando `def _commercial_sync_to_children()`, il prefetch su partner si limita al singolo partner e non viene esteso al resto dei `partners` risultanti dalla search.

Reply:
grazie 1K @astirpe per la dettagliata spiegazione!